src/cdomains/intDomain.ml

```diff
@@ -923,6 +923,461 @@ struct
+    This function could be then used for both join and meet operations with different parameter n: 1 for join, 2 for meet *)   
+  let combined_event_list lattice_op (xs: int_t event list)  =
+    let l = match lattice_op with `Join -> 1 | `Meet -> 2 in
+    let aux (interval_count, acc) = function
+      | Enter x -> (interval_count + 1, if (interval_count + 1) >= l && interval_count < l then (Enter x)::acc else acc)
+      | Exit x -> (interval_count - 1, if interval_count >= l && (interval_count - 1) < l then (Exit x)::acc else acc) 
+    in
+    List.fold_left aux (0, []) xs |> snd |> List.rev
+    
+  let rec events_to_intervals = function
+  | [] -> []
+  | (Enter x)::(Exit y)::xs  -> (x, y)::(events_to_intervals xs)
+  | _ -> failwith "Invalid events list"
+  
+  let remove_gaps (xs: t) = 
+    let aux acc (l, r) = match acc with
+      | ((a, b)::acc') when Ints_t.compare (Ints_t.add b (Ints_t.one)) l >= 0 -> (a, r)::acc'
+      | _ -> (l, r)::acc
+    in 
+    List.fold_left aux [] xs |> List.rev
+    
+  let canonize (xs: t) = 
+    interval_set_to_events xs |> 
+    combined_event_list `Join |> 
+    events_to_intervals
+    
+  let unary_op (x: t) op = match x with 
+    | [] -> []
+    | _ -> canonize (List.filter_map op x)
+
+  let binary_op ik (x: t) (y: t) op : t = match x, y with
+    | [], _ -> []
+    | _, [] -> []
+    | _::_, _::_ -> canonize (List.filter_map op (cartesian_product x y))
+
+  include Std (struct type nonrec t = t let name = name let top_of = top_of let bot_of = bot_of let show = show let equal = equal end)
+
+  let equal_to_interval i (a, b) = 
+    if a = b && b = i then 
+      `Eq 
+    else if Ints_t.compare a i <= 0 && Ints_t.compare i b <=0 then 
+      `Top 
+    else 
+      `Neq
+  
+  let equal_to i xs = match List.map (equal_to_interval i) xs with
+    | [] -> failwith "unsupported: equal_to with bottom"
+    | [`Eq] ->  `Eq 
+    | ys -> if List.for_all (fun x -> x = `Neq) ys  then `Neq else `Top  
+     
+  include Std (struct type nonrec t = t let name = name let top_of = top_of let bot_of = bot_of let show = show let equal = equal end)
+
+  let set_overflow_flag ~cast ~underflow ~overflow ik =
+    let signed = Cil.isSigned ik in
+    if !GU.postsolving && signed && not cast then
+      Goblintutil.svcomp_may_overflow := true;
+    let sign = if signed then "Signed" else "Unsigned" in
+    match underflow, overflow with
+    | true, true ->
+      M.warn ~category:M.Category.Integer.overflow ~tags:[CWE 190; CWE 191] "%s integer overflow and underflow" sign
+    | true, false ->
+      M.warn ~category:M.Category.Integer.overflow ~tags:[CWE 191] "%s integer underflow" sign
+    | false, true ->
+      M.warn ~category:M.Category.Integer.overflow ~tags:[CWE 190] "%s integer overflow" sign
+    | false, false -> assert false
+  
+  let norm ?(cast=false) ik = function 
```

Comment:
`norm` is maybe an add name, as this does not `norm` the interval set representation but rather works on individual intervals.